<a href="https://colab.research.google.com/github/genachiasson/PowerShell/blob/master/AirQuality.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [13]:
import pandas as pd
import numpy as np
import requests
import json
import matplotlib.pyplot as plt
import seaborn as sns

In [16]:
url = "https://data.cityofnewyork.us/api/views/c3uy-2p5r/rows.json"

response = requests.get(url)
data = response.text
parsed = json.loads(data)

formatted_data = []
for row in parsed['data']:
  if len(row) >= 16:
    column_names = [col['name'] for col in parsed['meta']['view']['columns'][8:24]]
    formatted_data.append(dict(zip(column_names, row[8:24])))

# Create the Pandas DataFrame from the formatted data.
df_sm = pd.DataFrame(formatted_data)

# Display the first few rows of the DataFrame.
df_sm.head()

Unique ID Indicator ID                                   Name  \
0    179772          640  Boiler Emissions- Total SO2 Emissions   
1    179785          640  Boiler Emissions- Total SO2 Emissions   
2    178540          365                Fine particles (PM 2.5)   
3    178561          365                Fine particles (PM 2.5)   
4    823217          365                Fine particles (PM 2.5)   

          Measure Measure Info Geo Type Name Geo Join ID  \
0  Number per km2       number         UHF42         409   
1  Number per km2       number         UHF42         209   
2            Mean       mcg/m3         UHF42         209   
3            Mean       mcg/m3         UHF42         409   
4            Mean       mcg/m3         UHF42         409   

            Geo Place Name          Time Period           Start_Date  \
0         Southeast Queens                 2015  2015-01-01T00:00:00   
1  Bensonhurst - Bay Ridge                 2015  2015-01-01T00:00:00   
2  Bensonhurst - Bay Ridge  Annual Average 2012  2011-12-01T00:00:00   
3         Southeast Queens  Annual Average 2012  2011-12-01T00:00:00   
4         Southeast Queens          Summer 2022  2022-06-01T00:00:00   

  Data Value Message  
0        0.3    None  
1        1.2    None  
2        8.6    None  
3        8.0    None  
4        6.1    None

In [15]:
df_sm.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18025 entries, 0 to 18024
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   Unique ID       18025 non-null  object
 1   Indicator ID    18025 non-null  object
 2   Name            18025 non-null  object
 3   Measure         18025 non-null  object
 4   Measure Info    18025 non-null  object
 5   Geo Type Name   18025 non-null  object
 6   Geo Join ID     18016 non-null  object
 7   Geo Place Name  18016 non-null  object
 8   Time Period     18025 non-null  object
 9   Start_Date      18025 non-null  object
 10  Data Value      18025 non-null  object
 11  Message         0 non-null      object
dtypes: object(12)
memory usage: 1.7+ MB


Notable Details:

Data indicates that there is 18025 rows and 12 columns.


A. Explain why this dataset was chosen:

  This dataset was chosen to identify the airquality in NYC location and how diffirent particles and emissions impact those areas. Measurement occur for specified time periods and could potentially offer a comparitive analysis of air quality between locations in highly populated areas.  

B. Develop three business questions you could potentially answer using this dataset and hypothesize what the answer to those questions might be within the context of the dataset:

1. Does the population of a particular area increase or decrease fine particles/ boiler emissions that have been measured?
2. Is there any correlation between "Fine Particles" and "Boiler Emissions" within certain locations?
3. Does the time and duration of the measurement make a difference in the data reported, such as displaying more outliers?

Find mor information on fine particles, here:
https://www.health.ny.gov/environmental/indoors/air/pmq_a.htm


Find more information o boiler emissions, here:
https://watmfg.com/watmfg23082016/wp-content/uploads/2016/09/CB-7435-Emissions-Guide.pdf


